charts/csi-secret-provider-class/templates/deployment.yaml

```
@@ -0,0 +1,46 @@
+{{- if .Values.dummyDeployment.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  {{ template "common.metadata" $ }}
```

The nesting seems wrong here:
```suggestion
{{ template "common.metadata" . }}
```
Also, there is no reason to pass `$` when `.` contains the same content.